### PR TITLE
Fix unused resources clean up to keep resources used in effects

### PR DESCRIPTION
**Pull request by D8H**
### Changes
- Extracted `Project::ExposeResources` into `ResourceExposer` and make it use `ProjectBrowserHelper`.
- Expose object and layer effect resources (unit tested).

### CODE REVIEW

File: Core/GDCore/IDE/Project/ArbitraryResourceWorker.cpp [hunk 1]
```diff
@@ -170,6 +152,48 @@ void ArbitraryResourceWorker::ExposeEmbeddeds(gd::String& resourceName) {
...
+  }
+  if (resourceType == "json") {
+    ExposeJson(resourceName);
+    return;
+  }
+  if (resourceType == "video") {
+    ExposeVideo(resourceName);
+    return;
+  }
+  gd::LogWarning("Unexpected resource type: " + resourceType + " for: " + resourceName);
```
Comment (ClementPasteau) [inline Core/GDCore/IDE/Project/ArbitraryResourceWorker.cpp:193]: should this be a LogError? This feels more important than just a warning to me

File: Core/tests/ArbitraryResourceWorker.cpp [hunk 2]
```diff
@@ -124,12 +128,215 @@ TEST_CASE("ArbitraryResourceWorker", "[common][resources]") {
...
+    gd::ResourceExposer::ExposeWholeProjectResources(project, worker);
     REQUIRE(worker.bitmapFonts.size() == 1);
     REQUIRE(worker.bitmapFonts[0] == "res3");
     REQUIRE(worker.images.size() == 1);
     REQUIRE(worker.images[0] == "res1");
     REQUIRE(worker.audios.size() == 1);
     REQUIRE(worker.audios[0] == "res4");
   }
+
+  SECTION("Can find resource usage in external events") {
```
Comment (ClementPasteau) [inline Core/tests/ArbitraryResourceWorker.cpp:140]: love the tests

**Review by ClementPasteau — COMMENTED** [review]
lgtm as the tests are clean, but I'll let 4ian approve this one on the logic side

File: Core/tests/ArbitraryResourceWorker.cpp (hunk 2 above)
Comment (4ian) [reply Core/tests/ArbitraryResourceWorker.cpp:140]: +1, nice and clear

**Review by 4ian — APPROVED** [review]

**Pull request merged**